decompiler/frontend/binaryninja/tagging.py

```diff
@@ -0,0 +1,32 @@
+import binaryninja
+from binaryninja import TagType
+from compiler_idioms.matcher import Matcher
+
+
+class CompilerIdiomsTagging:
+    TAG_SYMBOL = "⚙"
+
+    def __init__(self, binary_view: binaryninja.BinaryView, start: int):
+        self._bv = binary_view
+        self._path = self._bv.file.filename
+        self._function_start = start
```

Comment:
I don't think we need those class attributes as you pass through `self._bv` anyways.